### PR TITLE
Homework 9

### PR DESCRIPTION
**Pull request by BBlack96**

### CODE REVIEW

File: JSBasics/Homework-9/Book.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        this._author = author;
+        this._year = year;
+    }
+
+    get title() {
+        return this._title;
+    }
+
+    set title(newTitle) {
+        typeof newTitle === 'string'? this._title = newTitle : console.log("Name of book should be String")
```
Comment (sttaran) [inline JSBasics/Homework-9/Book.js:14]: makes sense but in this case single `if` would be more readable

Comment (sttaran) [reply JSBasics/Homework-9/Book.js:14]: same for other setters

File: JSBasics/Homework-9/Book.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    static findOldestBook(books){
+        let oldestBook = books[0];
+        for (let i = 0; i < books.length; i++){
+            if (books[i]._year < oldestBook._year){
+                oldestBook = books[i]
+            }
+        }
+        return oldestBook._title
+    }
```
Comment (sttaran) [inline JSBasics/Homework-9/Book.js:45]: this solution works but it would be better to use sort or reduce for it

**Review by sttaran — CHANGES_REQUESTED** [review]

File: JSBasics/Homework-9/Book.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        return this._title;
+    }
+
+    set title(newTitle) {
+        if (typeof newTitle === 'string') {
+            this._title = newTitle;
+        } else {
+            console.log("Name of book should be String");
+        }
+    }
```
Comment (sttaran) [inline JSBasics/Homework-9/Book.js:19]: make it easier
```js
        if (typeof newTitle !== 'string') {
             console.log("Name of book should be String");
            return
       }
       this._title = newTitle;

```

Comment (sttaran) [reply JSBasics/Homework-9/Book.js:19]: same for all setters

**Review by sttaran — CHANGES_REQUESTED** [review]

File: JSBasics/Homework-9/Book.js (hunk 3 above)
Comment (BBlack96) [reply JSBasics/Homework-9/Book.js]: @sttaran Done) ty https://github.com/BBlack96/aqa-advanced/pull/11/commits/ab4a6229813e48f5d7ed822ad37cea22d0335dcd

**Pull request merged**